### PR TITLE
Update link subscription behaviour

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/26977

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java [hunk 1]
```diff
@@ -84,6 +87,7 @@ public class BraveVpnPreferences extends BravePreferenceFragment implements Brav
     private ChromeBasePreference mSubscriptionStatus;
     private ChromeBasePreference mSubscriptionExpires;
     private ChromeBasePreference mServerHost;
+    private ChromeBasePreference mLinkSubscriptionPrefernce;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java:90]: typo `mLinkSubscriptionPrefernce`->`mLinkSubscriptionPreference`

**Review by tapanmodh — APPROVED** [review]
LGTM

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 13` using the following build(s):

```

```

Using the instructions via https://github.com/brave/brave-core/pull/10408#issuecomment-939396800, launched Brave using the Griffin staging sever as per the following:

image

Went through the following STR/Cases:

* ensured that `BraveVPNLinkSubscriptionAndroidUIStudy:Enabled` via `brave://version` after restarting the browser
* ensured that `Link purchase to your Brave account` isn't visible when there's no active subscription after enabling VPN
* ensured that `Link purchase to your Brave account` appears/is visible once a subscription is active (used monthly test card)
* ensured that `Link purchase to your Brave account` is removed once the subscription expires after ~35mins
  * note: as per @deeppandya, there needs to be an interaction with VPN before the `Link` UI is removed
* ensured that `Link purchase to your Brave account` is removed once the subscription has been manually cancelled
  * note: as per @deeppandya, there needs to be an interaction with VPN before the `Link` UI is removed
* ensured that the `Link purchase to your Brave account` was removed once a subscription expired and the following was done:
  * attempted to restore a expired subscription, attempted to switch regions when subscription expired
* ensured that tapping on the `Link purchase to your Brave account` UI opens `account.brave.com`

**`Launching Brave without a subscription and then subscribing and ensuring "Link" UI appears`**

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** 
---------------|---------------|---------------|---------------|---------------|---------------|---------------|---------------
Screenshot_20221130-173847 | Screenshot_20221130-174117 | Screenshot_20221130-174135 | Screenshot_20221130-174139 | Screenshot_20221130-174147 | Screenshot_20221130-174151 | Screenshot_20221130-174158 | Screenshot_20221130-175506

**`Ensuring that the "Link" UI is removed once the subscription has expired`**

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
---------------|---------------|---------------|---------------|---------------|---------------
Screenshot_20221130-182749 | Screenshot_20221130-182930 | Screenshot_20221130-183313 | Screenshot_20221130-184108 | Screenshot_20221130-184223 | Screenshot_20221130-210815

**Pull request merged**